packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/RenameDocumentTreeItemHandler.java

```diff
@@ -39,34 +39,45 @@
         return ExplorerDescriptionProvider.DOCUMENT_KIND.equals(treeItem.getKind());
     }
 
     @Override
-    public IStatus handle(IEditingContext editingContext, TreeItem treeItem, String newLabel) {
-        var optionalEditingDomain = Optional.of(editingContext)
-                .filter(IEMFEditingContext.class::isInstance)
-                .map(IEMFEditingContext.class::cast)
-                .map(IEMFEditingContext::getDomain);
-
-        if (optionalEditingDomain.isPresent()) {
-            var editingDomain = optionalEditingDomain.get();
-            ResourceSet resourceSet = editingDomain.getResourceSet();
-            URI uri = new JSONResourceFactory().createResourceURI(treeItem.getId());
+    public IStatus handle(IEditingContext editingContext, TreeItem treeItem, String newLabel, Tree tree) {
+        IStatus result = new Failure("Something went wrong while handling this rename action.");
+        var optionalTreeDescription = this.getTreeDescription(editingContext, tree);
+        if (optionalTreeDescription.isPresent()) {
+            var variableManager = new VariableManager();
+            variableManager.put(IEditingContext.EDITING_CONTEXT, editingContext);
+            variableManager.put(TreeDescription.ID, treeItem.getId());
+            var object = optionalTreeDescription.get().getTreeItemObjectProvider().apply(variableManager);
+            if (object instanceof Resource resource) {
+                resource.eAdapters().stream()
+                        .filter(ResourceMetadataAdapter.class::isInstance)
+                        .map(ResourceMetadataAdapter.class::cast)
+                        .findFirst()
+                        .ifPresent(adapter -> adapter.setName(newLabel));
 
-            List<Resource> resourcesToRename = resourceSet.getResources().stream()
-                    .filter(resource -> resource.getURI().equals(uri))
-                    .toList();
-            resourcesToRename.forEach(resource -> resource.eAdapters().stream()
-                    .filter(ResourceMetadataAdapter.class::isInstance)
-                    .map(ResourceMetadataAdapter.class::cast)
-                    .findFirst()
-                    .ifPresent(adapter -> adapter.setName(newLabel)));
-
-            return new Success(ChangeKind.SEMANTIC_CHANGE, Map.of());
+                result = new Success(ChangeKind.SEMANTIC_CHANGE, Map.of());
+            }
         }
-        return new Failure("Something went wrong while handling this rename action.");
+        return result;
+    }
+
+    private Optional<TreeDescription> getTreeDescription(IEditingContext editingContext, Tree tree) {
+        return this.representationDescriptionSearchService
+        .findAll(editingContext).values().stream()
+        .filter(TreeDescription.class::isInstance)
+        .map(TreeDescription.class::cast)
+        .filter(td -> Objects.equals(td.getId(), tree.getDescriptionId()))
+        .findFirst();
```

Comment:
Can be removed and replaced inline with a simple call to `IRepresentationDescriptionSearchService#findById` instead

Reply:
Done